tests/Doctrine/Tests/ORM/Functional/Ticket/DDC7561Test.php

```diff
@@ -0,0 +1,282 @@
+     * @var string
+     */
+    private $currencyIsoCode;
+
+    /**
+     * @var DDC7561ConfirmationStatus
+     * @Column(type="ddc_7561_withdraw_order_confirmation_status")
+     */
+    private $DDC7561ConfirmationStatus;
+
+    /**
+     * DDC7561WithdrawOrder constructor.
+     * @param int $id
+     * @param string $amount
+     * @param string $currencyIsoCode
+     */
+    public function __construct(int $id, string $amount, string $currencyIsoCode)
+    {
+        parent::__construct($id);
+        $this->amount = $amount;
+        $this->currencyIsoCode = $currencyIsoCode;
+        $this->DDC7561ConfirmationStatus = new DDC7561ConfirmationStatus('pending');
+    }
+
+    /**
+     * @return string
+     */
+    public function getAmount(): string
+    {
+        return $this->amount;
+    }
+
+    /**
+     * @return string
+     */
+    public function getCurrencyIsoCode(): string
+    {
+        return $this->currencyIsoCode;
+    }
+
+    /**
+     * @return DDC7561ConfirmationStatus
+     */
+    public function getDDC7561ConfirmationStatus(): DDC7561ConfirmationStatus
+    {
+        return $this->DDC7561ConfirmationStatus;
+    }
+}
+
+class DDC7561DDC7561ConfirmationStatusType extends Type
+{
+    /**
+     * @param mixed            $value
+     * @param AbstractPlatform $platform
+     *
+     * @return string
+     *
+     * @throws ConversionException
+     */
+    public function convertToDatabaseValue($value, AbstractPlatform $platform): string
+    {
+        if (!$value instanceof DDC7561ConfirmationStatus) {
+            throw new ConversionException(\sprintf(
+                'Value must be instance of "%s", instance "%s" given',
+                DDC7561ConfirmationStatus::class,
+                \is_object($value) ? \get_class($value) : \gettype($value)
+            ));
+        }
+
+        /* @var DDC7561ConfirmationStatus $value */
+        return $value->getValue();
+    }
+
+    /**
+     * @param mixed            $value
+     * @param AbstractPlatform $platform
+     *
+     * @return DDC7561ConfirmationStatus
+     *
+     * @throws ConversionException
+     */
+    public function convertToPHPValue($value, AbstractPlatform $platform): DDC7561ConfirmationStatus
+    {
+        try {
+            return new DDC7561ConfirmationStatus($value);
+        } catch (\Throwable $e) {
+            throw new ConversionException($e->getMessage(), $e->getCode(), $e);
+        }
+    }
+
+    /**
+     * @param array            $fieldDeclaration
+     * @param AbstractPlatform $platform
+     *
+     * @return string
+     */
+    public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $platform): string
+    {
+        return 'ddc_7561_withdraw_order_confirmation_status';
+    }
+
+    /**
+     * @return string
+     */
+    public function getName(): string
+    {
+        return 'ddc_7561_withdraw_order_confirmation_status';
+    }
+}
+
+class DDC7561ConfirmationStatus {
```

Comment:
Let's simplify the test a bit: let's make this a generic `DDC7561NonNullableValue`, which just rejects `null`